server/asset/common.go

```diff
@@ -98,6 +103,9 @@ type BlockUpdate struct {
 	Reorg bool
 }
 
+// RPCError is an error type used for unexpected backend RPC failures.
+type RPCError error
```

Comment:
Really a combination of `err != nil` and `errors.Is(err, asset.CoinNotFoundError)` would be enough, but having this type seemed like a good idea in case we want to catch unexpected errors from the RPC call in particular.